src/object/srv_obj_migrate.c

```diff
@@ -2201,20 +2202,39 @@
 	struct enum_unpack_arg	*arg = data;
 	struct migrate_one	*mo;
 	bool			merged = false;
+	bool			create_migrate_one = false;
+	uint32_t		parity_shard;
+	uint32_t		shard;
 	int			rc = 0;
 	int			i;
 
 	if (!daos_oclass_is_ec(&arg->oc_attr))
 		return migrate_one_create(arg, io);
 
+	/* If parity shard alive for this dkey, then ignore the data shard enumeration
+	 * from data shard.
+ 	 */
+	rc = obj_ec_parity_alive(arg->oh, io->ui_dkey_hash, &parity_shard);
+ 	if (rc < 0)
+ 		return rc;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs